### DISCUSSION TITLE
Bugs LibreVNA 1.5 alpha 2

### DISCUSSION THREAD

**Discussion by Marc3lN** [jankae/LibreVNA#212] [General]
Hi, as I wanted to test something with the compound device, some bugs came up and I wanted to post them here as a feedback.
(Windows Laptop, GUI 1.5.0-alpha.2)

Calibration:
(1MHz to 6GHz, 501Points, level -10dB)
whenever I want to make a calibration measurement (doesn't matter if compound or single Device) there appears a very high number as minimum frequency: 
"501 points from 179769313486231564793826746329356487460061771816610493194377291867566634083253736182911671780280864445928163680987122391750825462330354250895282439122322875506826024599142533926918074193061745122574500020189880363468340637347674643851875759782894318316386198487970256787451014597457079993094755057664GHz to 0 Hz"
If i change the number of points, it will change with the next measurement, the output of the min and max frequency are static and resistant to changes in the GUI. I also tried 1MHz to 10MHz:
![image](https://github.com/jankae/LibreVNA/assets/103591430/c88d1aef-3500-4159-bf3e-e15750e17cfe)

If I take all measurements I cant activate my calibration because "not all measurements have been taken" (i think its related to the problem above, i didnt try calibrationg with a previous Version and just loading the calibration).

Compound device:
While creating the compound device there was only the LibreVNA suggested that I was connected to at the moment.
to see both, its necessary to disconnect.

I hope this helps you improve the app, if there are any Questions about the bugs do not hesitate to ask

**Comment by jankae** [comment]
> 501 points from 17976931348623156479382674632935648746006177181661049319437729186756663408...

This bug should already by fixed (although not in the alpha.2 release). Please try the latest development version: https://github.com/jankae/LibreVNA/actions/runs/4620048778

> Compound device:
While creating the compound device there was only the LibreVNA suggested that I was connected to at the moment.
to see both, its necessary to disconnect.

I can't try it myself right now but this might be a limitation of the libusb implementation on Windows. By the way: you don't need any LibreVNA device to create a compound device. The serial numbers for the individual devices are just suggestions, you can simply enter any text you want into the field (of course it must match an actual serial number, otherwise you won't be able to connect the compound device). Or, as you already figured out, you can disconnect from the LibreVNA.

**Reply by Marc3lN** [reply]
Thank you for your response. I found another bug in the latest development version: 
I tried the retrieve_trace_date.py script.
Everytime I try to connect after the connection check (vna.query("*IDN?")) I get a response from the GUI:("LibreVNA-GUI"),
then the GUI is closing/ crashing
if i use the released v1.5.0 alpha 2 version it doesn't crash and i get the measurement back

**Reply by ren13ukr** [reply]
I am not sure I did use your "latest development version" link exactly, but download and try to use Alfa and Alfa2 versions 
Same trouble with calibration 
![Create_Default2SOLT](https://github.com/jankae/LibreVNA/assets/68022176/964fb63e-7698-4216-af70-a8d6fe58587d)
![calibration_marked](https://github.com/jankae/LibreVNA/assets/68022176/22f6e15d-da87-4f75-968a-5912bf33fef1)
![SOLT12](https://github.com/jankae/LibreVNA/assets/68022176/6d2208dc-192c-4cbe-aeea-e530fb08df81)
![Throue12](https://github.com/jankae/LibreVNA/assets/68022176/e5fc5a7c-d964-43e1-9d5a-b53d1a4c4d4b)

So, I came back to 1.4,0 GUI ver.

**Reply by jankae** [reply]
What does Help->About say? That will identify the exact version you are using

**Reply by ren13ukr** [reply]
1.5.0-alpha.2-73426459e

**Reply by jankae** [reply]
That is the 1.5.0-alpha.2 release from last November. The calibration frequency bug was fixed in 9776b81b which only happened at the beginning of this year. This means two things:

- I should probably do another release soon since the last one was quite some time ago
- You should either compile the GUI yourself or use one of the latest prebuilt binaries. They are available for every commit on the main branch, the latest being this one: https://github.com/jankae/LibreVNA/actions/runs/6785175567 (you can download the binaries on the bottom of that page if you are logged in to Github)

**Comment by jankae** [comment]
Sorry, I can't replicate that. When I use the latest commit and run the retrieve_trace_data script, everything works as expected.

LibreVNA Version (64 bit): 1.5.0-alpha.2-c18845fce
OS: Windows 11 Version 2009
CPU Arch: x86_64

Output of script:
```
LibreVNA-GUI
Connected to 206039903350
Setting up the sweep...
Waiting for the sweep to finish...
Reading trace data...
(2000000000.0, (-0.046199+0.635063j))
(2003000000.0, (-0.0377672+0.634144j))
(2006000000.0, (-0.0292842+0.633312j))
(2009000000.0, (-0.0220133+0.631291j))
(2012000000.0, (-0.0139123+0.630312j))
(2015000000.0, (-0.0053687+0.628684j))
(2018000000.0, (0.0026262+0.627451j))
(2021000000.0, (0.0110838+0.625964j))
(2024000000.0, (0.0194382+0.624305j))
(2027000000.0, (0.0276067+0.622767j))
(2030000000.0, (0.0360712+0.620915j))
(2033000000.0, (0.0438375+0.618885j))
(2036000000.0, (0.0521482+0.616758j))
(2039000000.0, (0.0602613+0.61449j))
...
```
